instrumentation/kotlin-coroutines/src/test/kotlin/KotlinCoroutineTests.kt

```diff
@@ -105,27 +104,17 @@ class KotlinCoroutineTests(private val dispatcher: CoroutineDispatcher) {
    */
   fun tracedWithDeferredFirstCompletions() = runTest {
 
-    val children = listOf(
-      async {
-        tracedChild("timeout1")
-        false
-      },
-      async {
-        tracedChild("timeout2")
-        false
-      },
-      async {
-        tracedChild("timeout3")
-        true
-      }
-    )
-
     withTimeout(TimeUnit.SECONDS.toMillis(30)) {
-      select<Boolean> {
-        children.forEach { child ->
-          child.onAwait { it }
-        }
-      }
+      tracedChild("timeout1")
```

Comment:
Why this change was required?

Reply:
The way I understand what this test does is, that it launches these coroutines and then `onAwait`s for the results from each with a timeout. `spotBugs` complains about passing null into https://github.com/open-telemetry/opentelemetry-java-instrumentation/pull/1346/files/a6525b40e700bf52a8dcf707cdad68bc0fa9bb91#diff-ca9202f7cbd3f2a56910810901063998L126 - it's obviously confused! Instead of adding the class to the exclusion, rewriting it this way keeps the original intent, makes the code somewhat simpler and satisfies spotBugs.

Reply:
I am not sure these are equivalent. First, old test waited 30s max in total, new test waits 30 seconds on each child.

And reading https://github.com/spotbugs/spotbugs/issues/573 I wonder if we should run spotbugs on Kotlin code at all :)

Reply:
`detekt` is definitely more appropriate.